packages/metro/src/node-haste/DependencyGraph.js

```diff
@@ -252,6 +252,7 @@ class DependencyGraph extends EventEmitter {
     const isPath =
```

Comment:
What essentially `isPath` is checking for?

Reply:
It's short-circuiting whether to use a "global" cache or a cache that's relative to the current path. At Facebook all module ids are global and engineers require them like `require('MyModule')` without a path. This code here optimizes the common use case of requiring a global module.